README.md

```diff
@@ -1,2 +1,230 @@
+
+
+Complete example: [examples/hyperparameter-tuning/main.py](./examples/hyperparameter-tuning/main.py)
+
+#### Population Based Training
+
+We can also ask `fairing` to train our code using [Population Based Training](https://deepmind.com/blog/population-based-training-neural-networks/).
+
+This is a more advanced training strategies that needs hook into different lifecycle steps of your model, thus we need to define several additional method into our model class.
+
+A multiple read/write PVC name needs to be pass to the `PopulationBasedTraining` strategie. This is used to store and exchange the different models generated by our training to enable the `explore/exploit` mechanism of Population Based Training.
+
+```python
+from fairing.train import Train
+from fairing.strategies.pbt import PopulationBasedTraining
+
+@Train(
+    repository='<your-repo-name>',
+    strategy=PopulationBasedTraining(
+        population_size=10,
+        exploit_count=6,
+        steps_per_exploit=5000,
+        pvc_name='<pvc-name>',
+        model_path = MODEL_PATH
+    )
+)
+class MyModel(object):
+    def hyperparameters(self):
+      # returns the dictionary of hyperparameters
+    
+    def build(self, hp):
+      # build the model
+    
+    def train(self, hp):
+      # training logic
+    
+    def save(self):
+      # save the model at MODEL_PATH
+    
+    def restore(self, model_path):
+      # restore the model from MODEL_PATH
+```
+
+Complete example: [examples/population-based-training/main.py](./examples/population-based-training/main.py)
+
+
+<!-- ### Training Architectures
+
+#### Basic Architure
+
+This is the default `architecture`, each training run will be a single container acting in isolation.
+No `architecure` is specified since this is the default value.
+
+```python
+# Note: we are note specifiying any architecture since this is the default value
+@Train(package={'name': '<your-image-name>', 'repository': '<your-repo-name>', 'publish': True})
+class MyModel(object):
+    ...
+```
+
+Complete example: [examples/simple-training/main.py](./examples/simple-training/main.py) -->
+
+## Usage with Kubeflow
+
+### Simple TfJob
+
+Instead of creating native `Jobs`, `fairing` can leverage Kubeflow's `TfJobs` assuming you have Kubeflow installed in your cluster.
+Simply pass the Kubeflow architecture to the train decorator (note that you can still use all the training strategies mentionned above):
+
+```python
+from fairing.train import Train
+from fairing.architectures.kubeflow.basic import BasicArchitecture
+
+@Train(repository='wbuchwalter', architecture=BasicArchitecture())
+class MyModel(object):
+    def train(self):
+       # training logic
+```
+
+
+### Distributed Training
+
+Using Kubeflow, we can also ask `fairing` to start [distributed trainings](https://www.tensorflow.org/deploy/distributed) instead.
+Simply import `DistributedTraining` architecture insteda of the `BasicArchitecture`:
```

Comment:
Shouldn't distributed training be the default? Why would you not want to run it distributed if you're running it on kubeflow?